libraries/AP_HAL/board/esp32.h

```diff
@@ -61,3 +67,99 @@
 #define CONFIG_ESP32_WIFI_TX_BA_WIN 0
 #define CONFIG_ESP32_WIFI_RX_BA_WIN 0
 
+
+// turn off all the compasses by default.. 
+#ifndef AP_COMPASS_DIAGONALS_ENABLED
+#define AP_COMPASS_DIAGONALS_ENABLED 0
+#endif
+#ifndef AP_COMPASS_BACKEND_DEFAULT_ENABLED
```

Comment:
This should be sufficient to turn all of the backends off that you don't want - you don't need to specify the others.